src/deroff.rs

```diff
@@ -1477,12 +1484,212 @@ fn test_request_or_macro() {
     assert!(deroffer.s.is_empty());
     assert_eq!(deroffer.output.take(), "Hello World");
 }
-
 #[test]
-fn test_deroff() {}
+fn test_deroff() -> io::Result<()> {
+    use diff::lines;
+    use git2::Repository;
+    use itertools::izip;
+    use std::{
+        ffi::OsString,
+        fs::{self, read_dir},
+        io::Write,
+        path::{Path, PathBuf},
+    };
+
+    const TEST_OUTPUT_DIR: &'static str = "./test_deroff_output";
+
+    // Clear previous test output
+    if Path::new(TEST_OUTPUT_DIR).exists() {
+        fs::remove_dir_all(TEST_OUTPUT_DIR)?;
+    }
+    // Create the output directory
+    fs::create_dir(TEST_OUTPUT_DIR)?;
+
+    let stdout = io::stdout();
+    let mut lock = stdout.lock();
+
+    // If we don't have the manpages, get them
+    if !Path::new("./manpages").exists() {
+        print!("Cloning remote manpages... ");
+        lock.flush()?;
+        match Repository::clone("https://github.com/rust-dc/manpages.git", "./manpages") {
```

Comment:
I don't think we should be doing this in our test. Maybe we could clone it first as a submodule.